### ISSUE TITLE
WorldCompass training may overwrite EMA old-model prediction in negative-aware loss

### ISSUE THREAD

**Issue by archernsy** [Tencent-Hunyuan/HY-WorldPlay#61]
Hi, thanks for releasing HY-WorldPlay and the WorldCompass training code.

I may have found a possible implementation issue in the WorldCompass training loss, and would appreciate your confirmation.

Upstream commit checked: `ad424f57cf1054521a24b609dc0832afd7cd1a50`

File:
`worldcompass/fastvideo/training/world_compass_in_train_pipeline.py`

Current code around the negative-aware fine-tuning loss:

```python
with torch.no_grad():
    with self.ema_generator.apply_policy_shadow_to_model(
        self.transformer
    ):
        model_pred_old = model_pred = self.transformer(
            txt_branch=False,
            input_dict=input_dict,
        )[0]

model_pred_old = model_pred = self.transformer(
    txt_branch=False,
    input_dict=input_dict,
)[0]
```

The second assignment appears to overwrite `model_pred_old` with the current model prediction. As a result, the EMA old-policy prediction computed inside `apply_policy_shadow_to_model()` is discarded.

This seems inconsistent with the WorldCompass paper. In Sec. 3.4 / Eq. 6, the loss uses separate predictions from the EMA old model and the current trainable model:

```text
v_theta+ = (1 - beta) * v_theta_old + beta * v_theta
v_theta- = (1 + beta) * v_theta_old - beta * v_theta
```

The paper also states:

```text
v_theta_old and v_theta are predictions from the old model for rollout sampling
and the current model undergoing training. Both models are initialized from the
base model, and the old model is EMA updated by the trained model.
```

With the current code, for `beta = 1`, the intended

```text
v_theta- = 2 * v_theta_old - v_theta
```

effectively becomes:

```text
v_theta- = 2 * stopgrad(v_theta) - v_theta
```

This still gives a signed gradient because of `detach()`, but it no longer uses the EMA old model as the reference/anchor described in the paper. It also makes the preceding EMA forward pass unused except for extra compute.

A possible fix may be:

```python
with torch.no_grad():
    with self.ema_generator.apply_policy_shadow_to_model(self.transformer):
        model_pred_old = self.transformer(
            txt_branch=False,
            input_dict=input_dict,
        )[0]

model_pred = self.transformer(
    txt_branch=False,
    input_dict=input_dict,
)[0]
```

Could you confirm whether the current overwrite is intentional, or whether `model_pred_old` should preserve the EMA old-model output here?
